pumps/sql_aggregate.go

```diff
@@ -119,6 +145,64 @@ func (c *SQLAggregatePump) Init(conf interface{}) error {
 	return nil
 }
 
+// ensureIndex creates the new optimized index for tyk_aggregated.
+// it uses CONCURRENTLY to avoid locking the table for a long time - postgresql.org/docs/current/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY
+// if background is true, it will run the index creation in a goroutine
+// if not, it will block until it finishes
+func (c *SQLAggregatePump) ensureIndex(tableName string, background bool) error {
+	if !c.db.Migrator().HasIndex(tableName, newAggregatedIndexName) {
+		createIndexFn := func(c *SQLAggregatePump) error {
+			option := ""
+			if c.dbType == "postgres" {
+				option = "CONCURRENTLY"
+			}
+
+			err := c.db.Table(tableName).Exec(fmt.Sprintf("CREATE INDEX %s IF NOT EXISTS %s ON %s (dimension, timestamp, org_id, dimension_value)", option, newAggregatedIndexName, tableName)).Error
```

Comment:
if `c.dbType` is different than "postgres", `option` would be an empty string and this will probably fail (didn't reproduce it) - is that expected?

Reply:
It doesn't fail since `CREATE INDEX IF NOT EXISTS ` is a valid query for other dbTypes.